### DISCUSSION TITLE
Authorization error (401)

### DISCUSSION THREAD

**Discussion by voloshyn** [pandodao/developers.pando.im#2] [4swap]
Hello Pando Team!
We are experiencing the issue with orders API returning authorization error (401) while using a valid auth token.
It has happened to 2 clients before, and after 6-7 days, the problem disappeared without any changes from our side. 
Another client is experiencing the problem now calling this API:  https://api.4swap.org/api/orders/b75cf53a-7c6b-488b-98e7-cea06688b477
Important to note that 4Swap APIs work, and we performed hundreds of successful swaps; this issue rarely appears and persists for that client only.

**Comment by lyricat** [comment]
Hi, could you provide more details about the issue? for example, how do you generate the auth token? is there any error message for the 401 error? is the token works to request `https://api.mixin.one/me`?

**Reply by voloshyn** [reply]
Hi @lyricat  
yes, here is how we generate the token:
```
final authToken = signAuthTokenWithEdDSA(
  session.userId,
  session.sessionId,
  session.privateKey,
  null,
  'GET',
  '/me',
  '',
);

final uri = '$_fourSwapBaseUrl/orders/$followId';

final response = await httpClient.get(
  Uri.parse(uri),
  headers: {'Authorization': 'Bearer $authToken'},
);

```
Yes, it works with mixin request, that's why we thought that the token is valid.

The body of the response is:
`{"code":401,"msg":"authentication required"}`

**Reply by voloshyn** [reply]
`signAuthTokenWithEdDSA` is from mixin_bot_sdk-0.3.0

**Reply by voloshyn** [reply]
One of the users who had this problem is from our team, and he shared his private keys with us to help debug. We generated the tokens in-app and using `mixin-cli` and reproduced the error on our local computers. Shortly after that, the problem disappeared without any changes.

**Reply by voloshyn** [reply]
This is the command I used: `curl -H "Authorization: Bearer $(mixin-cli sign /me -f user.json)" https://api.4swap.org/api/orders/USERS_ORDER_ID`

**Reply by lyricat** [reply]
Hmm. I see. We will discuss the issue and back to you soon.

**Reply by lyricat** [reply]
Have you considered reusing an existing token instead of generating a new one for every request?

**Reply by voloshyn** [reply]
We can reuse the token, that's easy to change. Did you locate the root cause of the problem? Will this change fix it for good?

**Reply by lyricat** [reply]
One potential explanation for this issue could be that a new token is generated each time and is marked as invalid when a 502 error responded by Mixin's API (it occurs randomly). To resolve the problem, reusing the token may be a possible solution.
